Improve deployment docs

gongjason opened the pull request:
Mostly condensing the information into 1 doc with references to detailed sections. It's much easier to deploy Firezone now than when we first wrote these docs.

gongjason committed:
simplify deployment instructions

gongjason committed:
fix lint error

gongjason committed:
add troubleshooting section

gongjason committed:
additional server instrucrtions

gongjason committed:
clarify supported distros in intro

gongjason committed:
punctuation

gongjason committed:
upgrade to 0.3.17

jamilbk commented on docs/docs/administer/upgrade.md:
```diff
@@ -25,6 +25,12 @@ To upgrade Firezone, follow these steps:
 Occasionally problems arise. If you hit any, please let us know by [filing an
 issue](https://github.com/firezone/firezone/issues/new/choose).
 
+## Upgrading from >= 0.3.11 to >= 0.3.17
```
* Did you mean `Upgrading from 0.3.x to >= 0.3.16` as `>= 0.3.11 to >= 0.3.17` is somewhat ambiguous?
* Still need to test this to be sure this is the case
* I think 0.3.16 introduced this

We also need to be very clear here (make it bold, warning label) that any OIDC users added pre-`0.3.16` will need to revoke their tokens for this functionality to work, or the OAuth app needs to be deleted and recreated in their provider's portal.

gongjason replied:
oops this is supposed to be WIP in the OIDC docs branch. Removed it here.

jamilbk commented on docs/docs/deploy/index.md:
```diff
@@ -9,10 +9,84 @@ description: >
-[Start Here: Supported Platforms]({%link docs/deploy/supported-platforms.md%}){:.btn.btn-purple}
+### Firewall settings
+
+Firezone requires ports `80/tcp`, `443/tcp`, and `51820/udp` to be open.
```
This sort of contradicts what we say on the security considerations page: https://docs.firezone.dev/docs/administer/security-considerations/

Also, I'd reword this to be clear that typically `80/tcp` or `443/tcp` need to be available, not both. We highly recommend `443/tcp` and by default will redirect http requests to be https.

Also these can change so should clarify "by default"

Also they'll need to make sure forwarding is enabled in their firewall -- see the debugging https://docs.firezone.dev/docs/administer/troubleshoot/#debugging-connectivity-issues

jamilbk commented on docs/docs/deploy/index.md:
```diff
@@ -9,10 +9,84 @@ description: >
+console instructions for logging in to the web UI.
+
+By default, the web UI can be reached at the IP or domain name of your server.
+You can re-generate the admin credentials using the
```
Typo -> `regenerate`

jamilbk commented on docs/docs/deploy/index.md:
```diff
@@ -9,10 +9,84 @@ description: >
+
+## Troubleshooting
+
+Check the [troubleshooting guide]({% link docs/administer/troubleshoot.md %}).
```
First, check our troubleshooting guide to see if your issue is covered there

jamilbk reviewed (changes requested):
Some content clarifications. Looks good, let's keep improving the quality of these docs!

gongjason committed:
clarifications from review

@jamilbk changed based on your comments. There was a section about upgrading that was supposed to be in another branch (OIDC). I removed that.

@lilibobear added a screenshot

gongjason committed:
add note for ssh interruptions during install

gongjason committed:
link to github issue for ssh interrupt

jamilbk commented on docs/docs/deploy/prerequisites.md:
```diff
@@ -34,8 +34,9 @@ to do so:
-By default, Firezone requires ports `80/tcp`, `443/tcp`, and `51820/udp` to be open.
-This can change based on what you've configured in the configuration file.
+By default, Firezone requires ports `80/tcp` or `443/tcp`, and `51820/udp` to be
+accessible. We highliy recommend using https over `443/tcp`.
```
Typo: `highliy`

jamilbk replied:
I think we can drop the `80/tcp` requirement as it's not *required* by default. For clarification, this could be

> By default, Firezone requires ports `443/tcp` and `51820/udp` traffic to be accessible for HTTPS and WireGuard traffic respectively.

jamilbk commented on docs/docs/deploy/index.md:
```diff
@@ -5,14 +5,112 @@ nav_order: 2
+Ensure port forwarding is enabled on your firewall.
+The default Firezone configuration requires the following ports to be open:
+
+* `80/tcp` or `443/tcp`: To access the web UI. We highly recommend https over `443/tcp`.
```
I would drop `80/tcp` here since we require `443/tcp` by default (we redirect 80 to 443 by default). You could then drop the `recommend https over 443/tcp` part -- might just be me but it reads like `we highly recommend https instead of 443/tcp`

jamilbk reviewed (changes requested):
Just a few more minor cleanups

gongjason committed:
review note fixes

gongjason committed:
testing signed commit

gongjason committed:
update gpg email
